scheduler/src/cook/mesos/util.clj

```diff
@@ -24,7 +24,59 @@
             [metatransaction.core :refer (db)]
             [metrics.timers :as timers]
             [plumbing.core :as pc :refer (map-vals map-keys)])
-  (:import [java.util Date]))
+    (:import
+      [com.google.common.cache Cache]
+      [com.google.common.cache CacheBuilder]
+      [java.util.concurrent TimeUnit]
+      [java.util Date]))
+
+
+(defn new-cache []
+  "Build a new cache"
+  (-> (CacheBuilder/newBuilder)
+      (.maximumSize 1000000)
+      ;; if its not been accessed in 2 hours, whatever is going on, its not being visted by the
+      ;; scheduler loop anymore. E.g., its probably failed/done and won't be needed. So,
+      ;; lets kick it out to keep cache small.
+      (.expireAfterAccess 2 TimeUnit/HOURS)
+      (.build)))
+
+(defn lookup-cache!
```

Comment:
We should report metrics on the hit rate per cache, it may need an extra argument to be passed in to name the metric.

Reply:
Track under https://github.com/twosigma/Cook/issues/739